tests/jenkins/TestOpenSearchReleasePromotionTest.groovy

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+import org.yaml.snakeyaml.Yaml
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+import static com.lesfurets.jenkins.unit.global.lib.GitSource.gitSource
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.hamcrest.CoreMatchers.hasItem
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.junit.jupiter.api.Assertions.assertThrows
+
+class TestOpenSearchReleasePromotionTest extends BuildPipelineTest {
+
+    @Override
+    @Before
+    void setUp() {
+
+        helper.registerSharedLibrary(
+            library().name('jenkins')
+                .defaultVersion('5.11.1')
+                .allowOverride(true)
+                .implicit(true)
+                .targetPath('vars')
+                .retriever(gitSource('https://github.com/opensearch-project/opensearch-build-libraries.git'))
+                .build()
+            )
+
+        super.setUp()
+        addParam('RELEASE_VERSION', '1.0.0')
+        addParam('OPENSEARCH_RC', '2050')
+        addParam('OPENSEARCH_DASHBOARDS_RC', '3050')
+        binding.setVariable('AGENT_LINUX_X64', 'Jenkins-Agent-AL2023-X64-C54xlarge-Docker-Host')
+    }
+
+    @Test
+    void shouldExecuteWithoutErrors() {
+        super.testPipeline('jenkins/promotion/release-promotion.jenkinsfile',
+                'tests/jenkins/jenkinsjob-regression-files/promotion/release-promotion.jenkinsfile')
+    }
```

Comment:
Can we add some assertions to what all stages should be present at this job? Since this is a single point of execution I believe strong assertions are good to maintain. Any small change should break the tests indicating release can be blocked because of this change.

Reply:
I have added multiple assertions for each stage and assertions with parameters as well, so that child promotion jobs are not triggered with wrong values when updating the jenkinsfile.